### PR TITLE
Add some tests for Version

### PR DESCRIPTION
**Pull request by hughbe**

### CODE REVIEW

File: src/System.Runtime/tests/System/Version.cs [hunk 1]
```diff
@@ -93,14 +93,18 @@ public static IEnumerable<object[]> CompareToTestData()
...
+
         yield return new object[] { new Version(1, 2, 3, 4), new Version(1, 2, 3, 4), 0 };
         yield return new object[] { new Version(1, 2, 3, 4), new Version(1, 2, 3, 5), -1 };
         yield return new object[] { new Version(1, 2, 3, 4), new Version(1, 2, 3, 3), 1 };
     }
 
     [Theory]
     [MemberData(nameof(CompareToTestData))]
-    public static void TestCompareTo(Version version1, Version obj, int expectedSign)
+    public static void TestCompareTo(Version version1, object obj, int expectedSign)
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/Version.cs]: Rather than change this from ```Version obj``` to ```object obj```, would it not be better to change it to ```Version version2``` and get rid of the first line in the test?

File: src/System.Runtime/tests/System/Version.cs [hunk 2]
```diff
@@ -192,7 +196,8 @@ private static IEnumerable<object[]> Parse_Valid_TestData()
         yield return new object[] { "1.2", new Version(1, 2) };
         yield return new object[] { "1.2.3", new Version(1, 2, 3) };
         yield return new object[] { "1.2.3.4", new Version(1, 2, 3, 4) };
-        yield return new object[] { "2  .3.    4.  \t\r\n15  ", new Version(2, 3, 4, 15) };
+        yield return new object[] { "   2  .3.    4.  \t\r\n15  ", new Version(2, 3, 4, 15) };
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/Version.cs]: Why did you add the spaces to the beginning?  Should this case be added as an additional input rather than replacing the existing one?

**Comment by stephentoub** [comment]
One comment, otherwise LGTM. Thanks.

**Comment by hughbe** [comment]
Thanks, addressed your feedback

**Comment by stephentoub** [comment]
Thanks. LGTM.
